be1-go/hub/organizer.go

```diff
@@ -214,68 +213,26 @@ func (o *organizerHub) createLao(publish message.Publish) error {
 }
 
 type laoChannel struct {
-	clientsMu sync.RWMutex
-	clients   map[*Client]struct{}
-
-	inboxMu sync.RWMutex
-	inbox   map[string]message.Message
-
-	// /root/<base64ID>
-	channel string
-
-	witnessMu sync.Mutex
-	witnesses []message.PublicKey
+	*baseChannel
 }
 
 func (c *laoChannel) Subscribe(client *Client, msg message.Subscribe) error {
-	log.Printf("received a subscribe with id: %d", msg.ID)
-	c.clientsMu.Lock()
-	defer c.clientsMu.Unlock()
-
-	c.clients[client] = struct{}{}
-
-	return nil
+	return c.baseChannel.Subscribe(client, msg)
```

Comment:
The nice thing about embedding is the struct that embeds a parent struct as the ability to call methods on the embedded struct. Have a look at https://play.golang.org/p/iNTjT13SaIx for an example.

Reply:
Thank you, I removed the methods in laoChannel.